### ISSUE TITLE
Disable or Limit Disk Cache

### ISSUE THREAD

**Issue by tenplus1** [brave/brave-browser#5311]
Is there a way to disable or limit the disk cache in Brave ?

e.g.

In Chromium I can add this line to /etc/chromium-browser/default to limit cache on my ssd:

CHROMIUM_FLAGS="--disk-cache-dir=/dev/null --disk-cache-size=1 --media-cache-size=1"

**Comment by gorbehnare** [comment]
I need this also. Brave is killing my SSD in Linux and it seems like the default mode for Brave. Even tried new profile in Brave, new user account, and tried going through the flags looking for anything with cache or any other keywords to no avail. Even as I'm writing this on this page right here as the only tab open Brave has used several MB of writes to my SSD.

**Comment by dark-penguin** [comment]
I'm also very concerned about writing cache to my SSD. This is currently the only thing preventing me from using Brave as my default browser.

**Comment by dark-penguin** [comment]
I think you could workaround this by linking `~/.cache/BraveSoftware` to `/dev/null` . However, I can't be sure it's working - it might be just silently creating a cache directory elsewhere (for example, in the profile). Could we please get a confirmation that this will effectively disable disk cache?

**Comment by demetera** [comment]
This is working for me (sometimes). Start browser with following arguments: 
brave --disable-application-cache --media-cache-size=1 --disk-cache-size=1

After update, file /usr/share/applications/brave-browser.desktop gets overwritten by default launcher

**Comment by dark-penguin** [comment]
> After update, file /usr/share/applications/brave-browser.desktop gets overwritten by default launcher

Not only that, but if you click on a link in another application while Brave is closed, it will open without those parameters. So commandline parameters are not the way to go. Removing the cache directory and blocking it from being recreated is most likely a better and permanent solution, unless Brave is trying to be sneaky, and creates it elsewhere instead.

However, launching Brave from a commandline, I see that it's throwing errors trying to access the cache directory, which likely means that the workaround works.

**Comment by demetera** [comment]
You forced me to find another more robust solution and I've just tried it and it worked!

1) I had 2Gb RAMdisk created long time ago (for Spotify cache) mounted under /mnt/ramdisk.
2) I've just deleted BraveSoftware folder in /home/user/.cache folder and created symbolic link with 
`ln -s /mnt/ramdisk BraveSoftware`
All cache go there now

**Comment by dark-penguin** [comment]
Yes, that also works, if you have enough RAM to spare. :)

**Comment by demetera** [comment]
I think it's worth to try with:

none /mnt/ramdisk tmpfs nodev,nosuid,noexec,nodiratime,size=10M 0 0

in /etc/fstab file. 
Could be, Brave will overwrite old cache with such a 10 Mb limitation :)
Didn't try it, but it might work!

**Comment by dark-penguin** [comment]
If you want to have cache in memory instead of the SSD, you can mount tmpfs of a specific size. If you want to disable disk cache completely, you can symlink /dev/null there as I did, instead of giving it a very small tmpfs size (and a symlink, and a mount).

**Comment by why-not-try-calmer** [comment]
> You forced me to find another more robust solution and I've just tried it and it worked!
> 
> 1. I had 2Gb RAMdisk created long time ago (for Spotify cache) mounted under /mnt/ramdisk.
> 2. I've just deleted BraveSoftware folder in /home/user/.cache folder and created symbolic link with
>    `ln -s /mnt/ramdisk BraveSoftware`
>    All cache go there now

That's an interesting tip, I've tried it, but `iotop` is still reporting writes. As far as I know `iotop` only reads from block devices, which suggests that the trick didn't work (at least not completely).

**Comment by why-not-try-calmer** [comment]
According to my `iotop` one of the most active process write to files under `~/.config/BraveSoftware` so symlinking that might be necessary.

**Comment by demetera** [comment]
>so apparently there are lots of writes to files under >~/.config/BraveSoftware. I am not sure how to solve >this issue as sending this to a ramdisk would destroy >profiles on every reboot.

Yes, but this folder contains no cache. Only browser history, configs, cookies etc. and amount of data shouldn't be so high comparing to .cache one

**Comment by why-not-try-calmer** [comment]
> > so apparently there are lots of writes to files under >~/.config/BraveSoftware. I am not sure how to solve >this issue as sending this to a ramdisk would destroy >profiles on every reboot.
> 
> Yes, but this folder contains no cache. Only browser history, configs, cookies etc. and amount of data shouldn't be so high comparing to .cache one

In theory yes, but you can look for yourself with `iotop`

**Comment by cesss** [comment]
But, did anybody succeed in getting a fixed cache limit? In my case, I'm willing to give Brave about 500MB of cache, but no more than that, and I'd like that it always keeps in cache the data from the most recent browsing sessions.

Now you can tell Brave to delete all the History on exit, but that's too radical: for example, I lose all the tabs images if I do that. Instead, I'd like to keep a bit of cache, at least the tab images.

**Comment by rkrisztian** [comment]
Firefox and LibreWolf allow you to:
* disable disk cache via advanced configuration (about:config),
* configure a memory cache (with fixed size or unlimited) without needing a ram disk for it.

Simple and easy. I need the same in Brave.

I agree that the command-line way is not the way to go.

**Comment by jardous** [comment]
yes. We need this. brave-browser is killing my SSD too

**Comment by alkeryn** [comment]
bump

**Comment by raldisk** [comment]
bump!
